### DISCUSSION TITLE
Unable to Make Prometheus Read the web-config.yml for TLS Encryption (Error: unknown long flag '--web.config.file')

### DISCUSSION THREAD

**Discussion by lijie123bes** [prometheus/prometheus#15010] [Q&A]
### What did you do?

I am following the [official documentation](https://prometheus.io/docs/guides/tls-encryption/) regarding the TLS encryption of Prometheus. My Spring Boot uses https endpoints and thus I want Prometheus to scrap them as is. I am following the instructions but I am having problem in configuring Prometheus to read my web-config.yml. Specifically the following command,

prometheus \
--config.file=/mnt/d/Projects/Msc-Thesis-Project/prometheus.yml \
--web.config.file=/mnt/d/Projects/Msc-Thesis-Project/web-config.yml \
--web.external-url=https://localhost:8443/

throws the following error,

Error parsing commandline arguments: unknown long flag '--web.config.file' prometheus: error: unknown long flag '--web.config.file'

### What did you see instead? Under which circumstances?

Error parsing commandline arguments: unknown long flag '--web.config.file' prometheus: error: unknown long flag '--web.config.file'

**Comment by roidelapluie** [comment]
What is the output of prometheus --version?

**Comment by Najeeb120** [comment]
Ensure that you are using compatible prometheus version that supports --web.config.file flag

**Comment by roidelapluie** [comment]
You don't need to use the `--web.config.file` flag; it's meant for configuring Prometheus' own web server, not for scraping HTTPS endpoints. Instead, you should focus on setting the `scheme: https` in your scrape configuration.

For example, in `your prometheus.yml`:

```yaml
scrape_configs:
  - job_name: 'spring-boot-app'
    scheme: https
    static_configs:
      - targets: ['localhost:8443']
    tls_config:
      insecure_skip_verify: true
```

This will allow Prometheus to scrape HTTPS endpoints directly. If you're using self-signed certificates or don't want to verify the certificates, adding the `insecure_skip_verify: true` under `tls_config` will skip certificate validation, which can be useful in development environments.

This should solve the issue without needing any additional web configuration for Prometheus itself.
